### DISCUSSION TITLE
JSON VS Tuple

### DISCUSSION THREAD

**Discussion by firstDismay** [hettie-d/NORM#4] [General]
Good afternoon.
I read your book on optimizing PostgreSQL it was useful.
Your variant of interaction via JSON seems to me interesting taking into account possible linking in REST ideology via PostgRESRT project possibilities and presence of self declared API (OpenAPI) it looks attractive. Currently my interface has about 700 functions, but I am working with the NPGSQL library, which allows to work with tuples of composite types on .Net5.X platform, but the need to interface with other platforms raised a lot of questions about serialization/deserialization of complex composite types. In the end I also settled on JSON but you have in principle a ready solution. Thank you.

**Comment by hettie-d** [comment]
Thank you so much for your feedback!
